### PR TITLE
Some improvements and bug fixes

### PR DESCRIPTION
**Pull request by ahmed605**
Changes:

- Added child mode

- Fix Arabic person server

- Fix English person server

### CODE REVIEW

File: Akinator.Api.Net/AkinatorClient.cs [hunk 1]
```diff
@@ -21,16 +21,18 @@ public class AkinatorClient : IAkinatorClient
...
         private readonly Language m_usedLanguage;
         private readonly ServerType m_usedServerType;
+        private readonly bool Child_Mode;
         private string m_session;
         private string m_signature;
         private int m_step;
         private int m_lastGuessStep;
 
-        public AkinatorClient(Language language, ServerType serverType, AkinatorUserSession existingSession = null)
+        public AkinatorClient(Language language, ServerType serverType, AkinatorUserSession existingSession = null, bool child_mode = false)
```
Comment (janniksam) [inline Akinator.Api.Net/AkinatorClient.cs]: Can you name it without an underscore? childMode

File: Akinator.Api.Net/AkinatorClient.cs [hunk 2]
```diff
@@ -21,16 +21,18 @@ public class AkinatorClient : IAkinatorClient
         private readonly HttpClient m_webClient;
         private readonly Language m_usedLanguage;
         private readonly ServerType m_usedServerType;
+        private readonly bool Child_Mode;
```
Comment (janniksam) [inline Akinator.Api.Net/AkinatorClient.cs]: m_childMode

File: Akinator.Api.Net/Utils/AkiUrlBuilder.cs [hunk 3]
```diff
@@ -6,16 +6,25 @@ namespace Akinator.Api.Net.Utils
...
+            var child_switch = String.Empty;
+            var question_filter = String.Empty;
+
             var server = ServerSelector.GetServerFor(language, serverType);
             if (string.IsNullOrEmpty(server))
             {
                 throw new InvalidOperationException($"No server does match the language {language} and server type {serverType}.");
             }
 
+            if (child_mode == true)
```
Comment (janniksam) [inline Akinator.Api.Net/Utils/AkiUrlBuilder.cs]: == true is redundant

File: Akinator.Api.Net/Utils/AkiUrlBuilder.cs [hunk 4]
```diff
@@ -6,16 +6,25 @@ namespace Akinator.Api.Net.Utils
...
+            {
+                child_switch = "true";
```
Comment (janniksam) [inline Akinator.Api.Net/Utils/AkiUrlBuilder.cs:22]: can you do it like
```
if (childMode)
{
   return $"WITH BOTH NEW PARAMETERS"
}

return $"THE OLD URL"
```

instead?

**Review by janniksam — CHANGES_REQUESTED** [review]

File: Akinator.Api.Net/Utils/AkiUrlBuilder.cs (hunk 4 above)
Comment (ahmed605) [reply Akinator.Api.Net/Utils/AkiUrlBuilder.cs:22]: Okay, but note that Akinator web app adds these parameters with no value with child mode disabled

Comment (janniksam) [reply Akinator.Api.Net/Utils/AkiUrlBuilder.cs:22]: Okay, I understand. Then feel free to keep it like you already had before.

**Review by janniksam — APPROVED** [review]

**Comment by ahmed605** [comment]
BTW I'm working on a new function to fetch all available servers from akinator server with language ISO code so we will not care about if any of these (manually written) servers goes outdated or not working anymore

**Comment by janniksam** [comment]
That is awesome. I am looking forward to reviewing it :) Might have some ideas for it aswell. Can you open a fresh branch for it?

**Comment by ahmed605** [comment]
@janniksam  Okay, I have found the endpoint, the required parameters and subject IDs (I mean the id for: Object, Animal and Person) and I'm writing the parser and the ISO codes enum, BTW do you have another way than enums it's very hard to write all language ISO codes in enum and class?

**Comment by ahmed605** [comment]
Also we can make the library choose the server randomly (of course in the range of the language), what's your opinion about this?

**Comment by janniksam** [comment]
I created to new issues for both "ideas".
I don't think the second one should be part of the library though. I have to think about it a bit more.

**Comment by janniksam** [comment]
Are you happy with this PR? I will complete it and push the new nuget then.

**Comment by ahmed605** [comment]
Yes I think this PR is fine now, and about the second one I left a reply in it's issue

**Pull request merged**